lib/controller.php

```diff
@@ -179,17 +181,28 @@ public function import_blob($blob) {
 			$body = wpmarkdown_markdown_to_html( $body );
 		}
 
+		$post = new WordPress_GitHub_Sync_Post( $args['ID'] );
+
 		// Can we really just mash everything together here?
+		$post_type = $post->get_type_from_path($path);
+		$post_name = $post->get_name_from_path($path);
```

Comment:
@WellingGuzman Hmmm, interesting. The only issue I'm thinking this may cause is for when we add draft support (#52); if we're using the directory `_drafts` for unpublished posts, would we import those as `post_type === 'draft'`?

I know I have an irrational disdain for regex's, but I prefer leaning on the YAML frontmatter as the "canonical" source of all the info about the post, and use the path as little as possible for that (which might mean we should refactor some other things). It would also mean, if you wanted to create a new post on GitHub and have it imported into WordPress, you'd have to add the YAML frontmatter yourself.

Which means, I *think* I'd rather say, if there is no `layout` attribute, we should just default it to `post_type = 'post'`. The other thing is it gives the user the freedom to drop the long prefix that often accompanies custom post types in plugins, so `_plugin_prefix_books` can just be `_books` if they want.

That said, I can probably be convinced otherwise if you think that's a better way. @benbalter thoughts on this?

Aside: I think there might be a bug in creating posts on GitHub; I'd have to test it, but I think if you create a post, then edit it on GitHub, without ever exporting it, the second edit would get imported as a new post because there wouldn't be a post ID on the YAML frontmatter. Kind of a corner case but maybe should be sorted.